File: twelve.js
function divisorsIn(num) {
	var count = 0
	for (var i = 1; i <= Math.floor(Math.sqrt(num)) - 1; i++) {
		if (num % i === 0) {
			count += 2
		}
	}
	if (Math.floor(Math.sqrt(num)) === Math.sqrt(num)) {
		count++
	}
	return count
}

var i = 0
var tri = 0
var val = 0
while (val < 501) {
	i++
	tri += i
	val = divisorsIn(tri)
}

console.log(tri)

File: twelve.rb
def divisors_in(num)

	root = Math.sqrt(num).to_i
	i = 1
	divs = 0
	while i < root
		if num % i === 0
			divs += 2
		end
		i += 1
	end
	if Math.sqrt(num) === root
		divs += 1
	end
	divs

end

found = false
i = 0
tri = 0
until found
	i += 1
	tri += i
	if divisors_in(tri) > 500
		found = true
	end
end

puts tri
